### ISSUE TITLE
Intel MKL ERROR: Parameter 3 was incorrect on entry to SGEBAL.

### ISSUE THREAD

**Issue by star0511** [jameschapman19/cca_zoo#37]
Data Is here https://github.com/star0511/cca_test

a issues  when I run dccae.py,it reports this error

total parameters: 201383936 ====> Epoch: 1 Average train loss: 20384.0430 ====> Epoch: 1 Average val loss: 12101.7695 Min loss 12101.77 ====> Epoch: 2 Average train loss: 50558.5391

Intel MKL ERROR: Parameter 3 was incorrect on entry to SGEBAL.

**Comment by jameschapman19** [comment]
I'll take a look at this later (aware you've closed). Essentially relates to #35. 

Usually this indicates:
optional(some zero eigenvalues in the whitened covariance matrix of the encodings) -> exploding gradients -> nans

Before updating to use pytorch 1.9 you would also get nans if two eigenvalues were very similar

Can be fixed sometimes by:
- changing the parameters r (regularisation) and eps (places a lower limit on regularisation). This can also mean reducing regularisation too
- increasing minibatch size (though you are using full batches in this case)
- reducing the number of latent dimensions
- I suspect (but not implemented yet) pretraining encoders might help - could write your own training function to do this the 'DeepWrapper' was written to have access to a one-liner

**Comment by star0511** [comment]
I tried to reduce the number of latent dimensions, but I still haven't solved this problem. Do you try to solve it?
